OSDOCS-13716: Documented the 4.18.6 z-stream RNs (#90797)
openshift/openshift-docs: release_notes/ocp-4-18-release-notes.adoc

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.18.6 --pullspecs
----

[id="ocp-4-18-6-bug-fixes_{context}"]
==== Bug fixes

// SME 
* Previously, Operator Marketplace used an older version, v1.24, of the `pod-security.kubernetes.io/` label. With this release, the namespace where Operator Marketplace is deployed now uses the Pod Security Admission (PSA) label marked as `latest`. (link:https://issues.redhat.com/browse/OCPBUGS-53149[*OCPBUGS-53149*])

* Previously, during cluster shutdown, a race condition prevented a stage OSTree deployment from finalizing if the deployment was moved to a staging location during a reboot operation. With this release, a fix removes the race condition from the OSTree deployment so that the staged deployment can finalize even during a reboot operation. (link:https://issues.redhat.com/browse/OCPBUGS-53111[*OCPBUGS-53111*])

> * Previously, the `audit-logs` container that handles the `SIGTERM` signal timed out. Kubelet needed to send a hard termination signal (SIGKILL) to the `audit-logs` container  to terminate the `SIGTERM` signal. With this release, a fix to a  process ID (PID) alias means that audit log can properly handle a `SIGTERM` signal without the signal timing out. (link:https://issues.redhat.com/browse/OCPBUGS-52982[*OCPBUGS-52982*])

Author: :heavy_check_mark:

> * Previously, the `apply-bootstrap` container did not handle a `SIGTERM` signal correctly. The container would wait for a sleep operation to complete before handling the signal which then exceeded the `termination-grace-period` of the pod. This situation required a `SIGKILL` signal to force the shutdown operation and allow the pods to finish deletion. With this release, the `apply-bootstrap` container now handles the signal `SIGTERM` correctly so a correct graceful shutdown period happens without the need for a `SIGKILLED` signal. (link:https://issues.redhat.com/browse/OCPBUGS-52878[*OCPBUGS-52878*])

Author: :heavy_check_mark:

* Previously, you could mirror an empty catalog during a mirror-to-disk operation and this caused the disk-to-mirror operation to fail. This empty catalog was generated from an invalid Operator entry in the `ImageSetConfiguration` CR. With this release, you can no longer mirror an empty catalog so a disk-to-mirror operation can succeed. (link:https://issues.redhat.com/browse/OCPBUGS-52943[*OCPBUGS-52943*])

> * When upgrading GCP clusters that use a boot disk that is not compatible with UEFI, shielded VM support cannot be enabled. Previously, this prevented the creation of new machines. With this release, shielded VM support is disabled for disks that are known to be incompatible with UEFI. This primarily affects customers upgrading from {product-title} version 4.12 to 4.13 using the GCP marketplace images. (link:https://issues.redhat.com/browse/OCPBUGS-52495[*OCPBUGS-52495*]) 

Author: :heavy_check_mark:

* Previously, node logs on the {product-title} web console did not close when you clicked outside the node logs menu. With this release, the node logs menu now closes when you click outside the node logs menu. (link:https://issues.redhat.com/browse/OCPBUGS-52490[*OCPBUGS-52490*]) 

// SME 

> * Previously, when you logged on to the Developer Sandbox from the {product-title} web console, the web console ignored the path in the URL and navigated the user to the `all projects` view on the Developer Sandbox instead of the namespace detailed in the URL. With this release, a fix corrects this behavior to the error no longer exists. (link:https://issues.redhat.com/browse/OCPBUGS-52406[*OCPBUGS-52406*]) 

Author: :heavy_check_mark:

> * Previously, the `capturegroup` inline diff algorithm in the `cluster-compare` tool failed to match the source text in an object with the capturegroup` regular expression from a reference template. This issue existed if the source text had a similar structure to a regular expression. With this release, a fix to the `capturegroup` inline diff algorithm means that this matching issue no longer occurs.(link:https://issues.redhat.com/browse/OCPBUGS-51306[*OCPBUGS-51306*])

Reviewer: ```suggestion
* Previously, the `capturegroup` inline diff algorithm in the `cluster-compare` tool failed to match the source text in an object with the `capturegroup` regular expression from a reference template. This issue existed if the source text had a similar structure to a regular expression. With this release, a fix to the `capturegroup` inline diff algorithm means that this matching issue no longer occurs. (link:https://issues.redhat.com/browse/OCPBUGS-51306[*OCPBUGS-51306*])
```

> * Previously, when running `oc-mirror` v2 on a continuous integration (CI) automation cycle and you viewed `oc-mirror` v2 logs on a non-TTY console, the output was missing progress information, because of an issue with the progress bar implementation. With this release, `oc-mirror` v2 now disables the progress bar implementation and uses plaintext logging instead for redirecting output so that the missing information no longer persists. (link:https://issues.redhat.com/browse/OCPBUGS-50996[*OCPBUGS-50996*])

Reviewer: ```suggestion
* Previously, when you ran `oc-mirror` v2 on a continuous integration (CI) automation cycle and you viewed `oc-mirror` v2 logs on a non-TTY console, the output was missing progress information because of an issue with the progress bar implementation. With this release, `oc-mirror` v2 now disables the progress bar implementation and uses plain text logging instead for redirecting output so that the missing information no longer persists. (link:https://issues.redhat.com/browse/OCPBUGS-50996[*OCPBUGS-50996*])
```
See https://redhat-documentation.github.io/supplementary-style-guide/#plain-text

Author: :heavy_check_mark:

> * Previously, when the cluster certificate authority (CA) bundle was updated with a custom CA bundle, a delay would occur for the change to reflect in the `insights-runtime-extractor` container. This issue occurred if the Insights Operator gathered data after the CA bundle is updated. The container might not have been restarted and the workload runtime data would not be gathered. With this release, a fix removes the delay so that this issue no longer occurs.(link:https://issues.redhat.com/browse/OCPBUGS-48790[*OCPBUGS-48790*])

Reviewer: ```suggestion
* Previously, if the Insights Operator gathered data after the cluster certificate authority (CA) bundle was updated with a custom CA bundle, a delay occurred for the change to be reflected in the `insights-runtime-extractor` container. As a consequence, the container might not be restarted, and the workload runtime data might not be gathered. With this release, a fix removes the delay so that this issue no longer occurs. (link:https://issues.redhat.com/browse/OCPBUGS-48790[*OCPBUGS-48790*])
```

Author: Note was removed from the advisory. Thanks for the review!
